content/docs/intro/concepts/programming-model.md

```diff
@@ -9,4529 +9,104 @@ menu:
 
-{{% /choosable %}}
-{{% choosable language go %}}
+**Stack**. [Stacks]({{< relref "/docs/intro/concepts/stack">}}) are an instantiation of a program and contain a collection of cloud resources. A project can have multiple stacks. Stacks are similar to environments. For example, a project might have a dev stack,  a test stack and a production stack.
 
-```go
-project := ctx.Project()
-stack := ctx.Stack()
-```
+**Resource**. A [resource]({{< relref "/docs/intro/concepts/resources">}}) represents a type of infrastructure, such as a compute instance. Resources can be aggregated together as component resources
 
-{{% /choosable %}}
-{{% choosable language csharp %}}
+**Resource Provider**. A [resource provider]({{< relref "/docs/intro/concepts/resource-providers">}}) A resource provider handles  communications  with a cloud service to create, read, update, and delete the resources you define in your Pulumi programs.
 
-```csharp
-var project = Deployment.Instance.ProjectName;
-var stack = Deployment.Instance.StackName;
-```
+**Inputs**. A resource has [inputs]({{< relref "/docs/intro/concepts/inputs-outputs">}}) that represent the properties and dependencies between resources. An example of an input is the connection string to a database.
 
-{{% /choosable %}}
+**Outputs**. A resource has [outputs]({{< relref "/docs/intro/concepts/inputs-outputs">}}) that are properties of the resource instance. Outputs are also how Pulumi tracks dependencies between resources.
 
-{{< /chooser >}}
+**Secrets**. One type of input is a [secret]({{< relref "/docs/intro/concepts/secrets">}}). Secrets are private data, such as passwords, used by a resource. Secrets can be encrypted.
 
-#### Logging
+**Environment Variables**. Another type of input are [environment variables]({{< relref "">}}), which the stack requires to deploy the infrastructure correctly. An example of an environment variable is the AWS region where you will deploy your infrastructure.
 
-The {{< pulumi-log >}} collection of functions allows you to log diagnostics, warnings, or errors with the Pulumi engine. These will be displayed alongside all other Pulumi output in the CLI and in the Pulumi Console. They will also be logged and kept for historical purposes if you ever want to use them to audit or diagnose what has transpired.
+**Stack Outputs**. A [stack output]({{< relref "/docs/intro/concepts/inputs-outputs">}}) consists of the properties that a stack exports, such as a domain name or a Kubernetes kubeconfig file.
 
-{{< chooser language "javascript,typescript,python,go,csharp" >}}
+**Stack References**. A [stack reference]({{< relref "/docs/intro/concepts/inputs-outputs">}}) connects the exported properties of one stack to another stack. For example, you can declare a Kubernetes cluster in one stack which consumes services in another stack.
 
-{{% choosable language javascript %}}
+### How the model works {#how-the-model-works}
```

Comment:
Per earlier comment, I think this section should be removed to keep the topic more focused.